presto-main/src/main/java/com/facebook/presto/cost/TableScanStatsRule.java

```diff
@@ -59,6 +59,10 @@ protected Optional<PlanNodeStatsEstimate> doCalculate(TableScanNode node, StatsP
         Constraint<ColumnHandle> constraint = new Constraint<>(node.getCurrentConstraint());
 
         TableStatistics tableStatistics = metadata.getTableStatistics(session, node.getTable(), ImmutableList.copyOf(node.getAssignments().values()), constraint);
+        if (tableStatistics.getRowCount().isUnknown()) {
```

Comment:
If no variable statistics, we could set the confidence to be false but not sure that would be effective